doubly_linked_list/doubly_linked_list.py

```diff
@@ -40,25 +40,66 @@ def __init__(self, node=None):
     self.tail = node
 
   def add_to_head(self, value):
-    pass
+    new_node = ListNode(value)
+    if self.head is None:
+      self.head = new_node
+    elif self.head is not None:
+      self.head.insert_before(value)
+      self.head = new_node
+
 
   def remove_from_head(self):
-    pass
+    if self.head is None:
+      return None
+    elif self.head is not None:
+      tmp = self.head
+      if self.head.next is not None:
+        self.head.delete()
+        self.head = self.head.next
+      return tmp.value
 
   def add_to_tail(self, value):
-    pass
+    new_node = ListNode(value)
+    if self.tail is None:
+      self.tail = new_node
+    elif self.tail is not None:
+      self.tail.insert_after(value)
+      self.tail = new_node
 
   def remove_from_tail(self):
-    pass
+    if self.tail is not None:
+      tmp = self.tail
+      self.tail.delete()
+      return tmp
 
   def move_to_front(self, node):
-    pass
+    current_node = self.head
+    while current_node is not None:
+      if current_node == node:
+        self.add_to_head(node.value)
+        node.delete()
+      current_node = current_node.next
+
+
 
   def move_to_end(self, node):
-    pass
+    current_node = self.head
+    while current_node is not None:
```

Comment:
also be careful of how you would go about using a while loop, while you could do it this way, there can be a more simplistic and easy way to go about moving to the end. Although this is a very creative approach Alejandro, god job!.